### DISCUSSION TITLE
Upgrading from zod 3.19.* causes weird behaviours

### DISCUSSION THREAD

**Discussion by Lewenhaupt** [colinhacks/zod#2871] [Q&A]
Upon upgrading from 3.19.* there is a lot of weird stuff going on with z.infer. Suddenly all properties of an object are optional even though none of them are defined as optional. I.e. the resolved type becomes 

```
{
  id?: string;
}
```

but should have been 

```
{
  id: string;
}
```
Would really like to upgrade to get coerce and some other nice things but I can't figure out why everything breaks :/

**Comment by JacobWeisenburger** [comment]
You must enable strict mode in your tsconfig.json.

https://github.com/colinhacks/zod#requirements

**Reply by Lewenhaupt** [reply]
Already have it set to strict unfortunately.

**Reply by JacobWeisenburger** [reply]
are you able to make a codesandbox or something that can reproduce the issue?
